data-prepper-plugins/grok-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepper.java

```diff
@@ -271,6 +263,27 @@ private void mergeCaptures(final Map<String, Object> original, final Map<String,
         }
     }
 
+    private void mergeCaptures(final Event event, final Map<String, Object> updates) {
+        for (final Map.Entry<String, Object> updateEntry : updates.entrySet()) {
+
+            if (!(event.containsKey(updateEntry.getKey())) || keysToOverwrite.contains(updateEntry.getKey())) {
+                event.put(updateEntry.getKey(), updateEntry.getValue());
+                continue;
+            }
+
+            if (event.isValueAList(updateEntry.getKey())) {
+                final List<Object> fieldList = event.getList(updateEntry.getKey(), Object.class);
```

Comment:
For variable naming, I do `fieldList` over `keyList`. However, you have a lot of variables with the `field` prefix. I think we should try to keep consistent with using `key` since that is what we have decided on so far. We should commit fully to `field` or `key` being the standard naming for Data Prepper.

Reply:
Staying with  `Key`. I think this should be `valueList`.

Thanks for the feedback. I will make sure this reads better